### ISSUE TITLE
Featue Request: View all routes/urls

### ISSUE THREAD

**Issue by pedromorgan** [gorilla/mux#186]
Is there a way to get a list of all routes.. The application has a few models and subrouters, so it be great to be able to see all these routes as part of the `dev` cycle.
eg a plain text of

```
/
/contact/{id}
/foo
/foo/{bar}.{ext}
```

**Comment by elithrar** [comment]
Use https://godoc.org/github.com/gorilla/mux#Router.Walk to walk the
Routes in the Router.

**Comment by shawnps** [comment]
@pedromorgan I threw a quick example together, hope this helps:

```
package main

import (
	"fmt"
	"net/http"

	"github.com/gorilla/mux"
)

func handler(w http.ResponseWriter, r *http.Request) {
	return
}

func main() {
	r := mux.NewRouter()
	r.HandleFunc("/", handler)
	r.HandleFunc("/products", handler)
	r.HandleFunc("/articles", handler)
	r.HandleFunc("/articles/{id}", handler)
	r.Walk(func(route *mux.Route, router *mux.Router, ancestors []*mux.Route) error {
		t, err := route.GetPathTemplate()
		if err != nil {
			return err
		}
		fmt.Println(t)
		return nil
	})
	http.Handle("/", r)
}
```

**Comment by shawnps** [comment]
```
➜  ~  go run foo.go
/
/products
/articles
/articles/{id}
```

**Comment by elithrar** [comment]
👌🏼

**Comment by shawnps** [comment]
@elithrar let me know if you think this would be useful to have in docs/README and I can add it in, otherwise maybe we can close this issue out if no response from @pedromorgan?

**Comment by elithrar** [comment]
@shawnps More examples are always good!

**Comment by shawnps** [comment]
@elithrar cool. PR is here, let me know if it can be made better:

https://github.com/gorilla/mux/pull/217

**Comment by elithrar** [comment]
Closed via #217. Thanks @shawnps!

**Comment by adelowo** [comment]
@pedromorgan if you want a visual display (in the cli), you might want to check out this repo https://github.com/adelowo/muxlist
